WordPressSharedTests/TestAnalyticsTracker.m

```diff
@@ -38,3 +40,31 @@ - (void)trackString:(NSString *)event withProperties:(NSDictionary *)properties
 
 
 @end
+
+void WPAnalyticsTestVerifyUnregistered(WPAnalyticsMethodBehaviorInvocation invocation) {
+    id trackerMock = OCMStrictClassMock([TestAnalyticsTracker class]);
+    id expectation = [trackerMock reject];
+    invocation(expectation);
+    [trackerMock verify];
+}
+
+void WPAnalyticsTestVerifyRegistered(WPAnalyticsMethodBehaviorInvocation invocation) {
+    id trackerMock = OCMStrictClassMock([TestAnalyticsTracker class]);
+    id expectation = [trackerMock expect];
+    invocation(expectation);
+    invocation(trackerMock);
+    [trackerMock verify];
+}
+
+void WPAnalyticsTestVerifyMultipleTrackers(WPAnalyticsMethodBehaviorInvocation invocation) {
+    id trackerMock = OCMStrictClassMock([TestAnalyticsTracker class]);
+    id trackerMock2 = OCMStrictClassMock([TestAnalyticsTracker class]);
+    id expectation = [trackerMock expect];
+    id expectation2 = [trackerMock2 expect];
+    invocation(expectation);
+    invocation(expectation2);
+    invocation(trackerMock);
+    invocation(trackerMock2);
+    [trackerMock verify];
+    [trackerMock2 verify];
+}
```

Comment:
Is the reason these have been defined in the Objective-C layer, although used only in the Swift / Quick code, that they use OCMock?

I noticed that `OCMStrictClassMock` is a macro and we could replace it with its definition like `let trackerMock = OCMockObject.mock(for: TestAnalyticsTracker.self)`. But then things fall apart because OCMock relies on Objective-C's dynamism, but in Swift the type system gets in the way. I was not able to call `verify()` on my Swift `trackerMock` instance.

Reply:
> Is the reason these have been defined in the Objective-C layer, although used only in the Swift / Quick code, that they use OCMock?
Yes. I'd like to keep the original tests, but I'm not familiar with OCMock, not sure if there is a way to move all these code into a Swift file.